lib/Dialect/Comb/CombFolds.cpp

```diff
@@ -1227,6 +1227,32 @@ LogicalResult XorOp::canonicalize(XorOp op, PatternRewriter &rewriter) {
   auto size = inputs.size();
   assert(size > 1 && "expected 2 or more operands");
 
+  // Check De Morgan's laws
+  if (op.isBinaryNot()) {
+    auto getNegations = [&](auto term) {
+      SmallVector<Value, 2> negations;
+      auto operands = term->getOperands();
+      negations.reserve(operands.size());
+      llvm::transform(operands, std::back_inserter(negations), [&](Value x) {
+        auto allOnes =
```

Comment:
Producing inverted terms is used several places, can you pull it out to a function?

Reply:
36a9135 adds `createOrFoldNot` following `createOrFoldSExt`, and converted most “create XorOp for NotOp” usage to `createOrFoldNot` except two:

1. Line 2190 of CombFolds.cpp: op.falseValue() is the all one constant, so we don't need to use createOrFoldNot.
2. Line 2924 of LowerToHW.cpp: the original author uses a custom template to build Op.